### PR TITLE
Slack App Integration.

### PR DESCRIPTION
**Pull request by Siddhanttimeline**
### Describe your changes:

Fixes <issue-number>

I worked on ... because ...

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/apps/AppResource.java [hunk 1]
```diff
@@ -723,6 +724,13 @@ public Response delete(
       throw new IllegalArgumentException(
           CatalogExceptionMessage.systemEntityDeleteNotAllowed(app.getName(), "SystemApp"));
     }
+
+    if (app.getScheduleType().equals(ScheduleType.Scheduled)
```
Comment (mohityadav766) [inline openmetadata-service/src/main/java/org/openmetadata/service/resources/apps/AppResource.java:728]: why are we doing this, in AppResource, such kind of calls should be delegated to a more generic function served by Native Application, we cannot have app specific behaviours handled like this in AppResource

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/apps/AppResource.java [hunk 2]
```diff
@@ -756,6 +764,13 @@ public Response delete(
       throw new IllegalArgumentException(
           CatalogExceptionMessage.systemEntityDeleteNotAllowed(app.getName(), "SystemApp"));
     }
+
```
Comment (mohityadav766) [inline openmetadata-service/src/main/java/org/openmetadata/service/resources/apps/AppResource.java:767]: remove this

Comment (mohityadav766) [reply openmetadata-service/src/main/java/org/openmetadata/service/resources/apps/AppResource.java:767]: handle it in app specific function overriden for the app

File: openmetadata-service/src/main/java/org/openmetadata/service/security/JwtFilter.java [hunk 3]
```diff
@@ -90,7 +90,9 @@ public class JwtFilter implements ContainerRequestFilter {
           "v1/users/password/reset",
           "v1/users/checkEmailInUse",
           "v1/users/login",
-          "v1/users/refresh");
+          "v1/users/refresh",
+          "v1/collate/apps/slack/callback",
```
Comment (mohityadav766) [inline openmetadata-service/src/main/java/org/openmetadata/service/security/JwtFilter.java:94]: this cannot be here , we need to think of other solution here

Comment (mohityadav766) [reply openmetadata-service/src/main/java/org/openmetadata/service/security/JwtFilter.java:94]: and don't add /collate there

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/apps/AppResource.java (hunk 1 above)
Comment (Siddhanttimeline) [reply openmetadata-service/src/main/java/org/openmetadata/service/resources/apps/AppResource.java:728]: Noted.

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/apps/AppResource.java (hunk 2 above)
Comment (Siddhanttimeline) [reply openmetadata-service/src/main/java/org/openmetadata/service/resources/apps/AppResource.java:767]: Noted.

File: openmetadata-service/src/main/java/org/openmetadata/service/security/JwtFilter.java (hunk 3 above)
Comment (Siddhanttimeline) [reply openmetadata-service/src/main/java/org/openmetadata/service/security/JwtFilter.java:94]: I want to expose those endpoints. I am validating them internally using Bolt. I will check of other way. Noted.

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/apps/AppResource.java (hunk 1 above)
Comment (Siddhanttimeline) [reply openmetadata-service/src/main/java/org/openmetadata/service/resources/apps/AppResource.java]: Provided the cleanup function in NativeApplication. Each app can have its own cleanup implementation. Overrode the cleanup fn in SlackApplication. Now, from AppResource I am calling the cleanup function on app uninstall.

File: openmetadata-service/src/main/java/org/openmetadata/service/security/JwtFilter.java (hunk 3 above)
Comment (Siddhanttimeline) [reply openmetadata-service/src/main/java/org/openmetadata/service/security/JwtFilter.java]: Updated.

**Review by mohityadav766 — APPROVED** [review]

**Pull request merged**